compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -235,6 +235,40 @@ impl<'a> Parser<'a> {
                 self.bump();
             }
 
+            // Look for PHP's `<>` and recover
+            if op.node == AssocOp::Less
+                && self.token.kind == token::Gt
+                && self.prev_token.span.hi() == self.token.span.lo()
+            {
+                let sp = op.span.to(self.token.span);
+                self.struct_span_err(sp, "invalid comparison operator `<>`")
+                    .span_suggestion_short(
+                        sp,
+                        "`<>` is not a valid comparison operator, use `!=`",
+                        "<>".to_string(),
+                        Applicability::MachineApplicable,
+                    )
+                    .emit();
+                self.bump();
+            }
+
+            // Look for C++'s `<=>` and recover
+            if op.node == AssocOp::LessEqual
+                && self.token.kind == token::Gt
+                && self.prev_token.span.hi() == self.token.span.lo()
+            {
+                let sp = op.span.to(self.token.span);
+                self.struct_span_err(sp, "invalid comparison operator `<=>`")
+                    .span_suggestion_short(
+                        sp,
+                        "`<=>` is not a valid comparison operator, use `std::cmp::Ordering`",
+                        "<=>".to_string(),
+                        Applicability::Unspecified,
```

Comment:
Same as with `<>`, but this fix would either have to be more involved or use placeholders. The suggestion should be something along the lines of `(lhs).ord(&rhs)` or `Cmp::ord(lhs, &rhs)`.

Reply:
Thank you. It makes sense to me now what this person meant: https://github.com/rust-lang/rust/pull/89871#discussion_r729001595